toqito/helper/bell_notation_conversions.py

```diff
@@ -0,0 +1,748 @@
+
+
+    Consider a behaviour (probability distribution) in CG notation (desc = [2, 2, 2, 2]):
+
+    .. math::
+        P_{CG} =
+        \begin{pmatrix}
+            1 & 0.5 & 0.5 \\
+            0.5 & 0.25 & 0.25 \\
+            0.5 & 0.25 & 0.25
+        \end{pmatrix}
+
+    Converting to FP notation:
+
+    >>> p_cg = np.array([[1, 0.5, 0.5], [0.5, 0.25, 0.25], [0.5, 0.25, 0.25]])
+    >>> desc = [2, 2, 2, 2]
+    >>> cg_to_fp(p_cg, desc, behaviour=True) # doctest: +NORMALIZE_WHITESPACE
+    array([[[[0.25, 0.25],
+             [0.25, 0.25]],
+            [[0.25, 0.25],
+             [0.25, 0.25]]],
+           [[[0.25, 0.25],
+             [0.25, 0.25]],
+            [[0.25, 0.25],
+             [0.25, 0.25]]]])
+
+    :param cg_mat: The matrix in Collins-Gisin notation.
+    :param desc: A list [oa, ob, ia, ib] describing the number of outputs (oa, ob) and inputs (ia, ib).
+    :param behaviour: If True, assume input is a behaviour (default: False, assume functional).
+    :return: The probability tensor V[a, b, x, y] in Full Probability notation.
+
+    .. note::
+        This function is adapted from the QETLAB MATLAB package function `CG2FP`.
+
+    References
+    ==========
+    .. bibliography::
+        :filter: docname in docnames
+
+    """
+    oa, ob, ia, ib = desc
+    v_mat = np.zeros((oa, ob, ia, ib))
+
+    def aindex(a: int, x: int) -> int:
+        """CG matrix row index for Alice's outcome `a` (0..oa-2) and input `x` (0..ia-1). Returns 1-based index."""
+        return 1 + a + x * (oa - 1)
+
+    def bindex(b: int, y: int) -> int:
+        """CG matrix col index for Bob's outcome `b` (0..ob-2) and input `y` (0..ib-1). Returns 1-based index."""
+        return 1 + b + y * (ob - 1)
+
+    if not behaviour:
+        # Functional case logic
+        k_term = cg_mat[0, 0] / (ia * ib) if ia > 0 and ib > 0 else 0
+        for x in range(ia):
+            for y in range(ib):
+                # Fill V[a, b, x, y] for a < oa-1, b < ob-1
+                for a in range(oa - 1):
+                    a_term = cg_mat[aindex(a, x), 0] / ib if ib > 0 else 0
+                    for b in range(ob - 1):
+                        b_term = cg_mat[0, bindex(b, y)] / ia if ia > 0 else 0
+                        v_mat[a, b, x, y] = k_term + a_term + b_term + cg_mat[aindex(a, x), bindex(b, y)]
+                # Fill V[a, ob-1, x, y] for a < oa-1 (last column for Bob)
+                for a in range(oa - 1):
+                    a_term = cg_mat[aindex(a, x), 0] / ib if ib > 0 else 0
+                    v_mat[a, ob - 1, x, y] = k_term + a_term
+                # Fill V[oa-1, b, x, y] for b < ob-1 (last row for Alice)
+                for b in range(ob - 1):
+                    b_term = cg_mat[0, bindex(b, y)] / ia if ia > 0 else 0
+                    v_mat[oa - 1, b, x, y] = k_term + b_term
+                # Fill V[oa-1, ob-1, x, y] (bottom-right corner)
+                v_mat[oa - 1, ob - 1, x, y] = k_term
+
+    else:  # behaviour == 1
```

Comment:
Hmm, so based on the comment, should we make sure that `behavior` is a Boolean instead of an integer? I know in MATLAB/QETLAB the convention is to use `0` and `1`, but I think using that same convention here might be confusing for Python users (and possibly subtly error-prone).

Reply:
As a follow-up, it seems like you do treat this as a Boolean, but the comment `# behaviour == 1` is just misleading, so perhaps let's opt to remove that.